velox/vector/tests/VectorTestBase.h

```diff
@@ -451,6 +452,63 @@ class VectorTestBase {
     return velox::test::VectorMaker::flatten(vector);
   }
 
+  // Convenience function to create a vector of type Map[K, ARRAY[K]].
+  // Supports null keys, and values and even null elements.
+  // Example:
+  //    createMapOfArraysVector<int64_t>(
+  //      {{{1, std::nullopt}},
+  //       {{2, O({4, 5, std::nullopt})}},
+  //       {{std::nullopt, O({7, 8, 9})}}});
+  template <typename T>
+  VectorPtr createMapOfArraysVector(
+      std::vector<std::map<
+          std::optional<T>,
+          std::optional<std::vector<std::optional<T>>>>> maps) {
+    std::vector<vector_size_t> mapSizes;
+    std::vector<std::optional<T>> keys;
+    std::vector<std::optional<std::vector<std::optional<T>>>> values;
+    for (auto& map : maps) {
+      mapSizes.push_back(map.size());
+      for (const auto& [key, value] : map) {
+        keys.push_back(key);
+        values.push_back(value);
+      }
+    }
+
+    auto mapValues = makeVectorWithNullArrays(values);
+    auto mapKeys = makeNullableFlatVector<T>(keys);
+    auto size = mapSizes.size();
+
+    auto offsets = AlignedBuffer::allocate<vector_size_t>(size, pool_.get());
+    auto sizes = AlignedBuffer::allocate<vector_size_t>(size, pool_.get());
+
+    auto rawOffsets = offsets->template asMutable<vector_size_t>();
+    auto rawSizes = sizes->template asMutable<vector_size_t>();
+
+    vector_size_t offset = 0;
+    for (vector_size_t i = 0; i < size; i++) {
+      rawSizes[i] = mapSizes[i];
+      rawOffsets[i] = offset;
+      offset += mapSizes[i];
+    }
+
+    return std::make_shared<MapVector>(
+        pool_.get(),
+        MAP(BIGINT(), ARRAY(BIGINT())),
+        nullptr,
+        size,
+        offsets,
+        sizes,
+        mapKeys,
+        mapValues);
+  }
+
+  template <typename T>
+  static std::optional<std::vector<std::optional<T>>> makeOptional(
```

Comment:
What't the motivation for introducing this method ? It seems identical to std::make_optional.